gnome-inform7.spec

```diff
@@ -4,32 +4,37 @@
 
 Name: gnome-inform7
 Version: 6M62
-Release: 1.fc22
+Release: 1.fc29
 
 URL: http://inform7.com/
 License: GPLv3
 
 Group: Development/Languages
 Source: http://inform7.com/download/content/6M62/gnome-inform7-6M62.tar.xz
 
+AutoReqProv: no
+
 # Build requirements:
 # Extra build tools
-BuildRequires: gettext
-BuildRequires: pkgconfig
-BuildRequires: xz-lzma-compat
+Requires: intltool
```

Comment:
I had to remove AutoReqProv because detected the libwebkitgtk dependency and adds both libwebkitgtk-1.0.so and libjavascriptcoregtk-1.0.so to the dependency list.  This has the  side effect of causing dnf to fail on package installation for gnome-inform7 because there is no package that available to address the dependencies in any modern dnf repository.  If there was a way to filter those two out from AutoReqProv then I think we could turn it back on.

Reply:
I see, I didn't realize it would include those in the requirements despite them not being available. In that case maybe we can move these back to BuildRequires but copy the Requires from the previous Inform package, omitting the missing dependencies?

Reply:
From playing with the system to figure out a work around, the best I can tell is that it looks like the rpm packager either scans the binary for deps or directly handshakes with libtool regarding dependencies if it performs AutoReqProv.  The only way I know how to see what AutoReqProv selected for deps is to dump them with rpm -qR {package-name}.  This generates a list of *.so the rpm needs but not the specific packages it requires.  Do you know an easy way to see the packages required by a rpm built using AutoReqProv?

Reply:
I _think_ (with about 75% confidence) that we don't need the actual list of packages required, I think you can just put these in the spec file directly, e.g. `Requires: libatk-1.0.so.0()(64bit)`

Reply:
Looks like you were 100% right :->  I was able to modify the RPM spec using the exact list of deps generated by AutoReqProv minus webkit and javascript core.  Interestingly, if you include Require:<lib> in the spec it seems to override AutoReqProv behavior.  So at this point all of the 'packages' are back to being BuildRequires and there is an explicit list of Requires: in the spec that refer directly to only required *.so.  The only package I did not specify exact *.so for is the Recommends: compat-libicu62.  Performed RPM testing on existing Fedora 29 HW installation, fresh OS install of Fedora 29 on VM, and fresh OS install of Fedora 30 on VM.  RPM loaded through software center without issue and launched gnome-inform7.  

With these final changes (including the additional commits above aeb71dd to b970325) I also performed DEB testing.  On 16.04 installation went fine and application can launch, create a new project etc.  But it looks like some of the changes for newdialog.ui broke 18.04 with the binary crashing after install when you attempt to create a new project (Inform7 IDE splash screen shows up initially just fine). This seg fault error **_only_** happens on 18.04 when you attempt to create a new Inform7 project.  ERROR ** create_new_builder: Error while building interface: newdialog.ui.  _**Reverting aeb71dd fixes this issue for 18.04.**_  

After reverting aeb71bb I rebuilt on both 16.04 and 18.04 and tested installation on existing OS running on HW and fresh OS install for both on VM.  When you build a binary package specific for each OS version the results are perfect with application installing fine and then starting up and allowing you to create a new project or open an existing project.  Tested the compiler on small and large inform files as well, no issues.  As far as getting a common deb for both 16.04 and 18.04, it looks like you have to build on 16.04 to make that happen.  Building the deb on 18.04 and then attempting to load an run on 16.04 causes some weird issues.